samples/brokers/http/Dockerfiles/grpc.broker

```diff
@@ -0,0 +1,51 @@
+ARG PLATFORM=amd64
+ARG CROSS_BUILD_TARGET=x86_64-unknown-linux-gnu
+ARG BROKER=http
+
+FROM ${PLATFORM}/rust:1.47 as build
+
+ARG BROKER
+
+RUN rustup component add rustfmt --toolchain 1.47.0-x86_64-unknown-linux-gnu
+
+RUN USER=root cargo new --bin ${BROKER}
+
+WORKDIR /${BROKER}
+
+COPY ./samples/brokers/${BROKER}/Cargo.toml ./Cargo.toml
+RUN cargo build \
+    --bin=standalone \
+    --release
+RUN rm ./src/*.rs
+RUN rm ./target/release/deps/${BROKER}*
+
+COPY ./samples/brokers/${BROKER} .
+
+RUN cargo build \
+    --bin=broker \
+    --release
+
+FROM amd64/debian:buster-slim
+
+ARG BROKER
+
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    ca-certificates \
+    libssl-dev \
+    openssl && \
+    apt-get clean
+
+COPY --from=build /${BROKER}/target/release/broker /broker
+
+LABEL org.opencontainers.image.source https://github.com/deislabs/akri
+
+# Expose port used by broker service
+EXPOSE 8084
+
+ENV SSL_CERT_FILE=/etc/ssl/certs/ca-certificates.crt
+ENV SSL_CERT_DIR=/etc/ssl/certs
+ENV RUST_LOG ${BROKER},akri_shared
+
+ENTRYPOINT ["/broker"]
```

Comment:
`Dockerfiles/grpc.broker` and `Dockerfiles/grpc.client` look almost identical.  if you added a COMPONENT arg (being set to either broker or client), it looks like you wouldn't need the duplicate code.

Reply:
It's annoying but I feel it's something of a limitation of Dockerfiles.

If I switch to parameterizing with `ARG foo` then I must:

+ use an `ENV bar=foo` so that I can `ENTRYPOINT ./${bar}`
+ switch to command forms of `ENTRYPOINT` or `CMD` too in order to get a shell to evaluate the env
+ and then I'd need to: `the-container ./bar --grpc_endpoint=...` because you can't add args on the `docker run`

Reply:
fair enough.  thanks for trying :)